cmd/node/main.go

```diff
@@ -1693,6 +1694,12 @@ func createShardCoordinator(
 		log.Info("starting as observer node")
 
 		selfShardId, err = processDestinationShardAsObserver(prefsConfig)
+		if err != nil {
+			return nil, "", err
+		}
+		if selfShardId == core.DisabledShardIDAsObserver {
```

Comment:
move this if inside the processDestinationShardAsObserver function.

Reply:
same as above